actionview/lib/action_view/template.rb

```diff
@@ -101,6 +101,54 @@ class Template
     # You can use +local_assigns+ in the sub templates to access the local variables:
     #
     #   local_assigns[:headline] # => "Welcome"
+    #
+    # Each key in +local_assigns+ is available as a partial-local variable:
+    #
+    #   local_assigns[:headline] # => "Welcome"
+    #   headline                 # => "Welcome"
+    #
+    # Since +local_assigns+ is a +Hash+, it's compatible with Ruby 3.1's pattern
+    # matching assignment operator:
+    #
+    #   local_assigns => { headline:, **options }
+    #   headline                 # => "Welcome"
+    #   options                  # => {}
+    #
+    # Pattern matching assignment also supports variable renaming:
+    #
+    #   local_assigns => { headline: title }
+    #   headline                 # => "Welcome"
+    #   title                    # => "Welcome"
+    #
+    # If a template refers to a variable that isn't passed into the view as part
+    # of the +locals: { ... }+ Hash, the template will raise an
```

Comment:
The shorthand notion doesn't work with spaces:
```
echo "+locals: { ... }+" | rdoc --pipe
<p>+locals: { … }+</p>
```
```suggestion
    # of the <tt>locals: { ... }</tt> Hash, the template will raise an
```